src/main/java/com/tabnine/TabNineCompletionContributor.java

```diff
@@ -32,6 +35,10 @@
 public class TabNineCompletionContributor extends CompletionContributor {
     private final TabNineGateway process = DependencyContainer.singletonOfTabNineGateway();
 
+    private int consecutiveRestarts = 0;
```

Comment:
why not AtomicInteger ?

Reply:
No need, it is not multithreaded

Reply:
If it was multithreaded, IntelliJ wasn't hang when the response didn't arrive (that is why we needed to implement timeouts...)

Reply:
oh, I thought you're modifying it in the submit... nevermind